src/net/connection/timeout_thread.rs

```diff
@@ -83,3 +96,10 @@ impl TimeoutThread {
         Ok(())
     }
 }
+
+impl Drop for TimeoutThread {
+    fn drop(&mut self) {
+        // Stop thread, and discard the error if it exists as it is irrelevant.
+        self.stop().ok();
```

Comment:
Can you log a `error!` here if the `stop` fails. That way the end user is notified in a best effort way.

Reply:
Done.